packages/code/src/Code.tsx

```diff
@@ -169,16 +338,63 @@ function Code({
     </Syntax>
   );
 
+  function handleScroll(e: React.UIEvent) {
+    const scrollWidth = e.currentTarget.scrollWidth;
+    const elementWidth = e.currentTarget.clientWidth;
+    const isScrollable = scrollWidth > elementWidth;
+
+    if (isScrollable) {
+      const scrollPosition = e.currentTarget.scrollLeft;
+      const maxPosition = scrollWidth - elementWidth;
+
+      if (scrollPosition > 0 && scrollPosition < maxPosition) {
+        dispatch({ type: 'scrollBoth' });
+      } else if (scrollPosition > 0) {
+        dispatch({ type: 'scrollLeft' });
+      } else if (scrollPosition < maxPosition) {
+        dispatch({ type: 'scrollRight' });
+      }
+    }
+  }
+
   if (!multiline) {
     return (
       <div className={wrapperStyle}>
         {renderedWindowChrome}
 
         <div
-          {...(rest as DetailedElementProps<HTMLDivElement>)}
-          className={wrapperClassName}
+          className={css`
+            display: flex;
+          `}
         >
-          {renderedSyntaxComponent}
+          <div
+            {...(rest as DetailedElementProps<HTMLDivElement>)}
+            className={wrapperClassName}
+            onScroll={e => {
+              handleScroll(e);
+            }}
```

Comment:
Rather than making a new closure just pass the function directly to the handler. 

```suggestion
            onScroll={handleScroll}
```

Reply:
Made this change earlier today in one place but not others. Thanks for catching it!